cant use nightly build of jdk with jbang

follow instructions at https://github.com/openjdk/jdk/blob/master/doc/building.md

then:

```
jbang jdk install --force 21 `pwd`
[jbang] [ERROR] Java version in given path: /Users/manderse/code/openjdk/jdk/build/macosx-aarch64-server-release/jdk is 2023 which does not match the requested version 21
[jbang] Run with --verbose for more details
```

the release file contains this:
```
cat release
SOURCE=".:git:4ffc8cc2169c"
IMPLEMENTOR="N/A"
JAVA_VERSION_DATE="2023-09-19"
JAVA_RUNTIME_VERSION="21-internal-adhoc.manderse.jdk"
OS_NAME="Darwin"
OS_ARCH="aarch64"
LIBC="default"
```

```
 bin/java --version
openjdk 21-internal 2023-09-19
OpenJDK Runtime Environment (build 21-internal-adhoc.manderse.jdk)
OpenJDK 64-Bit Server VM (build 21-internal-adhoc.manderse.jdk, mixed mode)
```

I tried doing ` jbang jdk install --force 2023 `pwd`` which works but then when using `//PREVIEW` I get:

```
jbang --enable-preview -m main UnnamedClassMainMethodLauncher.java
[jbang] Building jar...
error: invalid source release: 2023
Usage: javac <options> <source files>
use --help for a list of possible options
[jbang] [ERROR] Error during compile
[jbang] Run with --verbose for more details
```

--- a/src/main/java/dev/jbang/util/JavaUtil.java
+++ b/src/main/java/dev/jbang/util/JavaUtil.java
@@ -143,7 +143,7 @@ public class JavaUtil {
 	public static Optional<String> readJavaVersionStringFromReleaseFile(Path home) {
 		try (Stream<String> lines = Files.lines(home.resolve("release"))) {
 			return lines
-						.filter(l -> l.startsWith("JAVA_VERSION=") || l.startsWith("JAVA_RUNTIME_VERSION"))
+						.filter(l -> l.startsWith("JAVA_VERSION=") || l.startsWith("JAVA_RUNTIME_VERSION="))
 						.map(JavaUtil::parseJavaOutput)
 						.findAny();
 		} catch (IOException e) {

--- a/src/main/java/dev/jbang/util/JavaUtil.java
+++ b/src/main/java/dev/jbang/util/JavaUtil.java
@@ -143,7 +143,7 @@ public class JavaUtil {
 	public static Optional<String> readJavaVersionStringFromReleaseFile(Path home) {
 		try (Stream<String> lines = Files.lines(home.resolve("release"))) {
 			return lines
-						.filter(l -> l.startsWith("JAVA_VERSION"))
+						.filter(l -> l.startsWith("JAVA_VERSION=") || l.startsWith("JAVA_RUNTIME_VERSION"))
 						.map(JavaUtil::parseJavaOutput)
 						.findAny();
 		} catch (IOException e) {

--- a/src/test/java/dev/jbang/cli/TestJdk.java
+++ b/src/test/java/dev/jbang/cli/TestJdk.java
@@ -13,6 +13,8 @@ import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Arrays;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
 import java.util.function.Function;
 
 import org.junit.jupiter.api.BeforeEach;
@@ -192,6 +194,16 @@ class TestJdk extends BaseTest {
 
 		assertThat(result.result, equalTo(SUCCESS_EXIT));
 		assertThat(result.normalizedOut(), containsString("cache" + File.separator + "jdks" + File.separator + "11"));
+	}
+
+	@Test
+	void testJavaRuntimeVersion() throws Exception {
+		Arrays.asList(21).forEach(this::createMockJdkRuntime);
+
+		CaptureResult result = checkedRun(jdk -> jdk.javaEnv("21"));
+
+		assertThat(result.result, equalTo(SUCCESS_EXIT));
+		assertThat(result.normalizedOut(), containsString("cache" + File.separator + "jdks" + File.separator + "21"));
 	}
 
 	@Test
@@ -414,17 +426,33 @@ class TestJdk extends BaseTest {
 	}
 
 	private void createMockJdk(int jdkVersion) {
+		createMockJdk(jdkVersion, this::initMockJdkDir);
+	}
+
+	private void createMockJdkRuntime(int jdkVersion) {
+		createMockJdk(jdkVersion, this::initMockJdkDirRuntime);
+	}
+
+	private void createMockJdk(int jdkVersion, BiConsumer<Path, String> init) {
 		Path jdkPath = JBangJdkProvider.getJdksPath().resolve(String.valueOf(jdkVersion));
-		initMockJdkDir(jdkPath, jdkVersion + ".0.7");
+		init.accept(jdkPath, jdkVersion + ".0.7");
 		Path def = Settings.getCurrentJdkDir();
 		if (!Files.exists(def)) {
 			Util.createLink(def, jdkPath);
 		}
 	}
 
+
+	private void initMockJdkDirRuntime(Path jdkPath, String version) {
+		initMockJdkDir(jdkPath, version, "JAVA_RUNTIME_VERSION");
+	}
+
 	private void initMockJdkDir(Path jdkPath, String version) {
+		initMockJdkDir(jdkPath, version, "JAVA_VERSION");
+	}
+	private void initMockJdkDir(Path jdkPath, String version, String key) {
 		Util.mkdirs(jdkPath);
-		String rawJavaVersion = "JAVA_VERSION=\"" + version + "\"";
+		String rawJavaVersion = key + "=\"" + version + "\"";
 		Path release = jdkPath.resolve("release");
 		try {
 			Util.writeString(release, rawJavaVersion);

